irohad/maintenance/metrics.cpp

```diff
@@ -9,88 +9,144 @@
+    return 1;  // return non-zero to disable civetweb logger
+  };
+  // create an http server running on addr:port
+  exposer_ = std::make_shared<Exposer>(listen_addr_port_,
+                                       /*num_threads*/ 2,
+                                       &cvcbs);
+
+  // ask the exposer_ to scrape the registry_ on incoming HTTP requests
+  exposer_->RegisterCollectable(registry_, "/metrics");
+
+  auto &block_height_gauge = BuildGauge()
+                                 .Name("blocks_height")
+                                 .Help("Total number of blocks in chain")
+                                 .Register(*registry_);
+  auto &block_height = block_height_gauge.Add({});
+  block_height.Set(storage_->getBlockQuery()->getTopBlockHeight());
+
+  auto &peers_number_gauge =
+      BuildGauge()
+          .Name("peers_number")
+          .Help("Total number peers to send transactions and request proposals")
+          .Register(*registry_);
+  auto &number_of_peers = peers_number_gauge.Add({});
+  number_of_peers.Set(storage_->getWsvQuery()->getPeers()->size());
+
+  auto &domains_number_gauge = BuildGauge()
+                                   .Name("number_of_domains")
+                                   .Help("Total number of domains in WSV")
+                                   .Register(*registry_);
+  auto &domains_number = domains_number_gauge.Add({});
+  domains_number.Set(storage_->getWsvQuery()->countDomains().assumeValue());
+
+  auto &total_number_of_transactions_gauge =
+      BuildGauge()
+          .Name("total_number_of_transactions")
+          .Help("Total number of transactions in blockchain")
+          .Register(*registry_);
+  auto &total_number_of_transactions =
+      total_number_of_transactions_gauge.Add({});
+  total_number_of_transactions.Set(
+      storage_->getWsvQuery()->countTransactions().assumeValue());
+
+  auto &number_of_signatures_in_last_block_gauge =
+      BuildGauge()
+          .Name("number_of_signatures_in_last_block")
+          .Help("Number of signatures in last block")
+          .Register(*registry_);
+  auto &number_of_signatures_in_last_block =
+      number_of_signatures_in_last_block_gauge.Add({});
+
+  auto calc_diffs = [](shared_model::interface::Block const &block) {
+    int domains_diff = 0, peers_diff = 0;
+    using namespace shared_model::interface;
+    for (Transaction const &trx : block.transactions()) {
+      for (Command const &cmd : trx.commands()) {
+        domains_diff += cmd.is<CreateDomain>() ? 1 : 0;
+        peers_diff += cmd.is<AddPeer>() ? 1 : 0;
+        peers_diff -= cmd.is<RemovePeer>() ? 1 : 0;
+      }
     }
-  }).detach();
+    return std::tuple{domains_diff, peers_diff};
+  };
 
-  return {registry};
+  block_subscriber_ = SubscriberCreator<bool, BlockPtr>::template create<
+      EventTypes::kOnBlock,
+      SubscriptionEngineHandlers::kMetrics>(
+      [&, registry = this->registry_]  /// Values are stored in registry_, hold
```

Comment:
```suggestion
      [registry = std::weak_ptr(registry_)...]  /// Values are stored in registry_, hold
```
It's better to capture everything with weak_ptrs to avoid destruction issues.

Reply:
done.